File: src/lib/types/dependency.ts
export interface DayDependency {
	id: string;
	day_plan_id: string;
	depends_on_day_plan_id: string;
	dependency_type: 'prerequisite' | 'recommended' | 'related';
	minimum_score: number;
	created_at: string;
}

export interface CreateDependencyInput {
	day_plan_id: string;
	depends_on_day_plan_id: string;
	dependency_type: string;
	minimum_score: number;
}

export interface DependencyStatus {
	dependency_id: string;
	depends_on_day_plan_id: string;
	depends_on_title: string;
	depends_on_day_number: number;
	dependency_type: string;
	minimum_score: number;
	is_met: boolean;
	best_score: number | null;
	best_attempt_status: string | null;
}

File: src-tauri/src/db/models/dependency.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct DayDependency {
    pub id: String,
    pub day_plan_id: String,
    pub depends_on_day_plan_id: String,
    pub dependency_type: String,
    pub minimum_score: i32,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateDependencyInput {
    pub day_plan_id: String,
    pub depends_on_day_plan_id: String,
    pub dependency_type: String,
    pub minimum_score: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DependencyStatus {
    pub dependency_id: String,
    pub depends_on_day_plan_id: String,
    pub depends_on_title: String,
    pub depends_on_day_number: i32,
    pub dependency_type: String,
    pub minimum_score: i32,
    pub is_met: bool,
    pub best_score: Option<i32>,
    pub best_attempt_status: Option<String>,
}
